### PR TITLE
Port window-normal-size

### PR DESCRIPTION
**Pull request by th0114nd**
https://github.com/remacs/remacs/issues/1358

### CODE REVIEW

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -713,6 +713,40 @@ pub extern "C" fn decode_any_window(window: LispObject) -> LispWindowRef {
...
+/// horizontal combination equals 1.0.
+///
+/// The normal sizes of windows are used to restore the proportional sizes
+/// of windows after they have been shrunk to their minimum sizes; for
+/// example when a frame is temporarily made very small and afterwards gets
+/// re-enlarged to its previous size.
+#[lisp_fn(min = "0")]
+pub fn window_normal_size(
+    window: LispWindowValidOrSelected,
+    horizontal: LispObject,
```
Comment (shaleh) [inline rust_src/src/windows.rs]: `horizontal` could be a boolean. The if/else below becomes `if !horizontal.....` or you could flip the if/else and use `if horizontal`.
It is not always safe to use a LispObject as a boolean. But on this case its only use is the nil check.

File: test/rust_src/src/windows-tests.el [hunk 2]
```diff
@@ -95,3 +95,13 @@
   (set-window-parameter (selected-window) 'test 'test)
   (should (consp (window-parameters)))
   (should (consp (window-parameters (selected-window)))))
+
```
Comment (shaleh) [inline test/rust_src/src/windows-tests.el]: Excellent. Thank you for the tests.

File: rust_src/src/windows.rs [hunk 3]
```diff
@@ -713,6 +713,40 @@ pub extern "C" fn decode_any_window(window: LispObject) -> LispWindowRef {
...
+) -> EmacsDouble {
+    let win: LispWindowRef = window.into();
+    EmacsDouble::from(if horizontal.is_nil() {
+        win.normal_lines
+    } else {
+        win.normal_cols
+    })
+}
```
Comment (shaleh) [inline rust_src/src/windows.rs]: I might have written this in the following style.

```
let value = if horizontal {
win.normal_cols
} else {
win.normal_lines
};
value.into() // or EmacsDouble::from(value)

File: rust_src/src/windows.rs [hunk 4]
```diff
@@ -26,8 +26,8 @@ use crate::{
         windows_or_buffers_changed, wset_redisplay,
     },
     remacs_sys::{
-        face_id, glyph_matrix, glyph_row, pvec_type, vertical_scroll_bar_type, EmacsInt, Lisp_Type,
-        Lisp_Window,
+        face_id, glyph_matrix, glyph_row, pvec_type, vertical_scroll_bar_type, EmacsDouble,
+        EmacsInt, Lisp_Type, Lisp_Window,
```
Comment (shaleh) [inline rust_src/src/windows.rs]: We usually try to import types separately from functions. This is all done to cut down on merge conflicts.

File: test/rust_src/src/windows-tests.el (hunk 2 above)
Comment (th0114nd) [reply test/rust_src/src/windows-tests.el]: NP

File: rust_src/src/windows.rs (hunk 3 above)
Comment (th0114nd) [reply rust_src/src/windows.rs]: Done

File: rust_src/src/windows.rs (hunk 4 above)
Comment (th0114nd) [reply rust_src/src/windows.rs]: I think these are types, but I separated the remacs binding ypes from the c struct types

**Comment by brotzeit** [comment]
:+1:

**Pull request merged**